wire-kotlin-generator/src/main/java/com/squareup/wire/kotlin/KotlinGenerator.kt

```diff
@@ -778,47 +807,54 @@ class KotlinGenerator private constructor(
         .build())
   }
 
-  private fun getDefaultValue(field: Field): CodeBlock {
-    val internalClass = ClassName("com.squareup.wire.internal", "Internal")
-    return when {
-      isEnum(field) -> if (field.default != null) {
-        CodeBlock.of("%T.%L", typeName(field.type()), field.default)
-      } else {
-        CodeBlock.of("null")
-      }
-      field.isRepeated -> CodeBlock.of("%T.newMutableList()", internalClass)
-      else -> CodeBlock.of("%L", field.default)
-    }
-  }
-
-  private fun isEnum(field: Field): Boolean = schema.getType(field.type()) is EnumType
+  private val Field.isEnum: Boolean
+    get() = schema.getType(type()) is EnumType
 
   private fun Field.getDeclaration(allocatedName: String) = when {
-    isRepeated -> CodeBlock.of("val $allocatedName = mutableListOf<%T>()", typeName(type()))
-    type().isMap -> CodeBlock.of("val $allocatedName = mutableMapOf<%T, %T>()",
-        typeName(type().keyType()), typeName(type().valueType()))
-    else -> CodeBlock.of("var $allocatedName: %T = %L", declarationClass(), getDefaultValue(this))
+    isRepeated -> CodeBlock.of("val $allocatedName = mutableListOf<%T>()", type().typeName)
+    isMap -> CodeBlock.of("val $allocatedName = mutableMapOf<%T, %T>()",
+        keyType.typeName, valueType.typeName)
+    else -> CodeBlock.of("var $allocatedName: %T = %L", declarationClass, getDefaultValue())
   }
 
-  private fun Field.declarationClass(): TypeName {
-    val fieldClass = getClass()
-    if (isRepeated || default != null) return fieldClass
-    return fieldClass.copy(nullable = true)
-  }
+  private val Field.declarationClass: TypeName
+    get() = when {
+      isRepeated || default != null -> getClass()
+      else -> getClass().copy(nullable = true)
+    }
 
   private fun Field.getClass(baseClass: TypeName = nameToKotlinName.getValue(type())) = when {
     isRepeated -> List::class.asClassName().parameterizedBy(baseClass)
     isOptional && default == null -> baseClass.copy(nullable = true)
     else -> baseClass.copy(nullable = false)
   }
 
+  private fun Field.getDefaultValue(): CodeBlock {
+    return when {
+      isRepeated && !javaInterOp -> CodeBlock.of("emptyList()")
```

Comment:
Some of these cases look repetitive, would that work better with nested conditionals?
```kotlin
return when {
  isRepeated -> {
    if (javaInterOp) {
      ...
    } else {
      ...
    }
  }
}
```